### PR TITLE
fix(#630): Move config values from property directly onto parameters ...

### LINKED ISSUE #630: invalid JSON: got EOF while reading request body when executing CommitContainerChanges
**Output of `dotnet --info`:**

```
.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.203\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.408 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**What version of Docker.DotNet?:**

```
3.125.14
```

**Steps to reproduce the issue:**
Execute CommitContainerChanges with Docker Desktop for Windows, Version 4.19.0 (106363)
with the following parameters:
- ContainerID: Id of a running container
- RepositoryName: mcr.microsoft.com/mssql/server
- Tag: 2019-CU15-ubuntu-20.04-20230502040827

**What actually happened?:**
image

**What did you expect to happen?:**
New image is created from running container

**Additional information:**
I feel like it worked up until the last Docker Desktop update (this is only a hunch however).

### PR DESCRIPTION
**Pull request by kevinboss**
Fixes #630

Ideally the parameter class would be redesigned to look similar to CreateContainerParameters but that would be a breaking change.

### CODE REVIEW

**Comment by kevinboss** [comment]
@dotnet-policy-service agree

File: src/Docker.DotNet/Endpoints/ImageOperations.cs [hunk 1]
```diff
@@ -255,9 +255,7 @@ public async Task<CommitContainerChangesResponse> CommitContainerChangesAsync(Co
                 throw new ArgumentNullException(nameof(parameters));
             }
 
-            var data = parameters.Config == null
-                ? null
-                : new JsonRequestContent<Config>(parameters.Config, this._client.JsonSerializer);
+            var data = new JsonRequestContent<Config>(parameters.Config ?? new Config(), this._client.JsonSerializer);
```
Comment (HofmeisterAn) [inline src/Docker.DotNet/Endpoints/ImageOperations.cs:258]: I do not believe this is the appropriate fix. As I mentioned in the related issue, we should generate the class `CommitContainerChangesParameters` (via [specgen](https://github.com/dotnet/Docker.DotNet/tree/master/tools/specgen)) similar to [CreateContainerParameters](https://github.com/dotnet/Docker.DotNet/blob/a0eba61f7eb0ebec1e52f38c3b91c28b54128a6f/src/Docker.DotNet/Models/CreateContainerParameters.Generated.cs) and remove the nested type. Then we can simply call:

```suggestion
            var data = new JsonRequestContent<CommitContainerChangesParameters>(parameters, this._client.JsonSerializer);
```

Comment (kevinboss) [reply src/Docker.DotNet/Endpoints/ImageOperations.cs]: I have no updated modeldefs and regenerated the classes.

**Review by HofmeisterAn — COMMENTED** [review]
👍 I think that is better, thanks! @galvesribeiro WDYT?

Also note what I found in the latest Moby [release notes](https://github.com/moby/moby/releases/tag/v24.0.1):

> API: Fix an issue where `POST /commit` did not accept an empty request body. https://github.com/moby/moby/pull/45568

**Review by galvesribeiro — APPROVED** [review]
LGTM

**Pull request merged**